### ISSUE TITLE
gpg keyserver timed out

### ISSUE THREAD

**Issue by VVCepheiA** [docker-library/mysql#49]
I think there might be a problem with the new keyserver. In Dockerfile Step 3
```sh
Step 3 : RUN apt-key adv --keyserver pool.sks-keyservers.net --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
 ---> Running in 2c694a391b75
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --secret-keyring /tmp/tmp.t4dFVjxGv1 --trustdb-name /etc/apt//trustdb.gpg --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-security-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-stable.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-squeeze-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-squeeze-stable.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-wheezy-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-wheezy-stable.gpg --keyserver pool.sks-keyservers.net --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
gpg: requesting key 5072E1F5 from hkp server pool.sks-keyservers.net
gpg: keyserver timed out
gpg: keyserver receive failed: keyserver error
INFO[0120] The command [/bin/sh -c apt-key adv --keyserver pool.sks-keyservers.net --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5] returned a non-zero code: 2 
```

When I changed it to  
```sh
RUN apt-key adv --keyserver pool.sks-keyservers.net:80 --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
```
based on [this link](http://unix.stackexchange.com/questions/75892/keyserver-timed-out-when-trying-to-add-a-gpg-public-key) it gives 
```sh
Step 3 : RUN apt-key adv --keyserver pool.sks-keyservers.net:80 --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
 ---> Running in 71918756993d
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --secret-keyring /tmp/tmp.BpCv0sNqRg --trustdb-name /etc/apt//trustdb.gpg --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-security-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-jessie-stable.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-squeeze-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-squeeze-stable.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-wheezy-automatic.gpg --keyring /etc/apt/trusted.gpg.d//debian-archive-wheezy-stable.gpg --keyserver pool.sks-keyservers.net:80 --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
gpg: requesting key 5072E1F5 from pool.sks-keyservers.net:80
gpgkeys: no keyserver host provided
gpg: keyserver internal error
gpg: keyserver receive failed: keyserver error
INFO[0000] The command [/bin/sh -c apt-key adv --keyserver pool.sks-keyservers.net:80 --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5] returned a non-zero code: 2
```

Thanks in advance.

**Comment by md5** [comment]
Looks like you might want to try `p80.pool.sks-keyservers.net:80` to test that particular workaround, per [this document](https://sks-keyservers.net/overview-of-pools.php).

**Comment by yosifkit** [comment]
`pool.sks-keyservers.net` is a pool of servers and I don't think all the servers in the pool are listening for requests on port 80. I think @md5's workaround could work for you.  Is there a reason for building it and not just doing `docker pull mysql:5.7`?  You can then easily extend the image in your own Dockerfile with `FROM mysql:5.7`.

**Comment by VVCepheiA** [comment]
Thanks for your answers @md5 @yosifkit 
`pool.sks-keyservers.net` doesn't work but based on the document you provide, `hkp://p80.pool.sks-keyservers.net:80`.. kind of work. 
```sh
Step 3 : RUN gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
 ---> Running in a116f9000984
gpg: directory `/root/.gnupg' created
gpg: new configuration file `/root/.gnupg/gpg.conf' created
gpg: WARNING: options in `/root/.gnupg/gpg.conf' are not yet active during this run
gpg: keyring `/root/.gnupg/secring.gpg' created
gpg: keyring `/root/.gnupg/pubring.gpg' created
gpg: requesting key 5072E1F5 from hkp server p80.pool.sks-keyservers.net
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key 5072E1F5: public key "MySQL Release Engineering <mysql-build@oss.oracle.com>" imported
gpg: no ultimately trusted keys found
gpg: Total number processed: 1
gpg:               imported: 1
```
It imported something but it failed Step 7
```sh
Step 7 : RUN {         echo mysql-community-server mysql-community-server/data-dir select '';         echo mysql-community-server mysql-community-server/root-pass password '';         echo mysql-community-server mysql-community-server/re-root-pass password '';         echo mysql-community-server mysql-community-server/remove-test-db select false;     } | debconf-set-selections     && apt-get update && apt-get install -y mysql-server="${MYSQL_VERSION}"* && rm -rf /var/lib/apt/lists/*     && rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql
 ---> Running in eb314735e76f
Get:1 http://security.debian.org wheezy/updates Release.gpg [836 B]
Get:2 http://http.debian.net wheezy Release.gpg [1655 B]
Get:3 http://http.debian.net wheezy-updates Release.gpg [836 B]
Get:4 http://repo.mysql.com wheezy Release.gpg [173 B]
Get:5 http://security.debian.org wheezy/updates Release [102 kB]
Get:6 http://http.debian.net wheezy Release [168 kB]
Get:7 http://repo.mysql.com wheezy Release [9670 B]
Ign http://repo.mysql.com wheezy Release
Get:8 http://http.debian.net wheezy-updates Release [124 kB]
Get:9 http://http.debian.net wheezy/main amd64 Packages [7625 kB]
Get:10 http://repo.mysql.com wheezy/mysql-5.7-dmr amd64 Packages [2748 B]
Get:11 http://security.debian.org wheezy/updates/main amd64 Packages [302 kB]
Get:12 http://http.debian.net wheezy-updates/main amd64 Packages [4104 B]
Fetched 8341 kB in 2s (3652 kB/s)
Reading package lists...
W: GPG error: http://repo.mysql.com wheezy Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 8C718D3B5072E1F5
```
However, I think yosifkit's suggestion makes sense, I should probably just use `FROM mysql:5.7`. I /*desperately*/ tried to build this mainly because I kept getting jdbc connection errors and failing to link mysql with tomcat, but I guess building it absolutely wouldn't help...

Thanks again.
